core/shared/src/main/scala/scalaz/zio/internal/tracing/ZIOFn.scala

```diff
@@ -0,0 +1,36 @@
+package scalaz.zio.internal.tracing
+
+import scalaz.zio.{ UIO, ZIO }
+
+/**
+ * Marks a ZIO utility function that wraps a user-supplied lambda.
+ *
+ * Allows access to the underlying lambda so that stacktraces can
+ * point to supplied user's code, instead of ZIO internals.
+ *
+ * NOTE: it being an `abstract class`, not trait is important as typecasing
```

Comment:
```suggestion
 * NOTE: it being an `abstract class`, not trait is important as typecasting
```

Reply:
`type casing` / `type case` is actually a valid term for matching on types, that was not a typo!

Reply:
TIL 😄 
Sorry I didn't know and spellchecker in IDEA underlined it.